.github/actions/setup-and-login/action.yml

```diff
@@ -0,0 +1,67 @@
+name: Setup Node, Env and login to GHCR
+
+inputs:
+  install-packages:
+    required: false
+    default: 'false'
+  build-and-run-server:
+    required: false
+    default: 'false'
+  ghcr-username:
+    required: false
+    default: ''
+  ghcr-password:
+    required: false
+    default: ''
```

Comment:
So it *doesn't* login if these aren't provided  - presumably for jobs which don't need to push. Would it not be a little cleaner to have two actions, one for the shared node + env and one for the docker login?